src/uu/mktemp/src/mktemp.rs

```diff
@@ -429,9 +419,16 @@ pub fn uu_app() -> Command {
                 )
                 .value_name("SUFFIX"),
         )
+        // Short option needed to handle the behavior of short/long options for TMPDIR.
+        // See Options trait for more information.
         .arg(
-            Arg::new(OPT_TMPDIR)
+            Arg::new(OPT_SHORT_TMPDIR)
                 .short('p')
+                .value_name("DIR")
```

Comment:
I think that we could add 
```suggestion
                .value_name("DIR")
                .overrides_with(OPT_TMPDIR)
```

(or add the inverse to OPT_TMPDIR, and then we don't need a lot of the complexity around figuring out which option comes last (because clap takes care of it for us).

Reply:
Yeah I like that approach. We could continue with you PR and copy the tests over from this PR.